tests/robustness/validate/patch_history.go

```diff
@@ -24,7 +24,26 @@ import (
 	"go.etcd.io/etcd/tests/v3/robustness/report"
 )
 
+func patchFailedRequestWithInfiniteReturnTime(reports []report.ClientReport) []report.ClientReport {
+	ret := make([]report.ClientReport, 0, len(reports))
+	for _, report := range reports {
+		operations := make([]porcupine.Operation, 0, len(report.KeyValue))
+		for _, operation := range report.KeyValue {
+			// Failed writes can still be persisted, setting to infinite for now as we don't know when request has took effect.
+			if operation.Output.(model.MaybeEtcdResponse).Error != "" {
+				operation.Return = math.MaxInt64
+			}
+			operations = append(operations, operation)
+		}
+		report.KeyValue = operations
+		ret = append(ret, report)
+	}
+
+	return ret
+}
+
 func patchLinearizableOperations(reports []report.ClientReport, persistedRequests []model.EtcdRequest) []porcupine.Operation {
+	reports = patchFailedRequestWithInfiniteReturnTime(reports)
```

Comment:
Like mentioned before, please don't call the function patch, nor put it in `patchLinearizableOperations`

Reply:
We should have something like:
``` 
func ValidateAndReturnVisualize(...) {
	err := checkValidationAssumptions(reports, persistedRequests)
	require.NoErrorf(t, err, "Broken validation assumptions")
	linearizableOperations, serializableOperations := prepareAndCategorizeOperations(reports)
	linearizableOperations = patchLinearizableOperations(linearizableOperations, reports, persistedRequests)
	...
}

func prepareAndCategorizeOperations(reports) (linearizable, serializable) {
	operations := patchFailedRequestWithInfiniteReturnTime(reports)
	return categorizeOperations(operations)
}

func patchFailedRequestWithInfiniteReturnTime()
```

Reply:
Becuase now we persist true return time and patch it before calling `patchLinearizableOperations`, the unit test will fail. 

The root cause is https://github.com/etcd-io/etcd/blob/27336678946c98bf272460e5dfc54909ecafc63e/tests/robustness/validate/patch_history.go#L218 will not work like before as in the unit test it will not have infinite set as return time for failed responses anymore, causing the unit tests to fail (actually, returning incorrect results for tests like `failed put remains if there is a matching persisted request, uniqueness allows for revision and return time to be based on watch`)